modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/source-builder/SourceBuilder.js

```diff
@@ -37,36 +37,46 @@ export default function SourceBuilder() {
 	);
 
 	useEffect(() => {
-		if (currentEditor?.mode === 'source' && elements) {
-			const metadata = {
-				description: definitionDescription,
-				name: definitionName,
-				version,
-			};
-
-			const xmlContent = serializeDefinition(
-				xmlNamespace,
-				metadata,
-				elements.filter(isNode),
-				elements.filter(isEdge)
-			);
-
-			if (xmlContent) {
-				currentEditor.setData(xmlContent);
-
-				setLoading(false);
+		function loadXmlContent() {
+			if (currentEditor?.mode === 'source' && elements) {
+				const metadata = {
+					description: definitionDescription,
+					name: definitionName,
+					version,
+				};
+
+				const xmlContent = serializeDefinition(
+					xmlNamespace,
+					metadata,
+					elements.filter(isNode),
+					elements.filter(isEdge)
+				);
+
+				if (xmlContent) {
+					currentEditor.setData(xmlContent);
+
+					setLoading(false);
+				}
 			}
 		}
 
+		const interval = setInterval(() => {
+			if (currentEditor) {
+				if (currentEditor?.mode !== 'source') {
```

Comment:
This optional chaining (?.) is not necessary because you just checked for `currentEditor` in the line before.

Reply:
Hi @cgmonte. I verified that is possible to decrease the delay, so now I put it to 1 second, which apparently is enough to setMode work correctly. I would like to keep the setInterval in the code to ensure if anything doesn't work as expected, this action will repeat until the setMode is changed.
Also, I did the other alteration requested for you.